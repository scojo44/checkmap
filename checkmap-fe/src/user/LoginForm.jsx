import React from 'react'
import {Link} from 'react-router-dom';
import {useForm} from 'react-hook-form'
import FormField from '../widgets/FormField';
import PasswordField from '../widgets/PasswordField';
// import './LoginForm.css'

export default function LoginForm({login}) {
  const {register, handleSubmit, formState: {errors}} = useForm();

  return (
    <form className="LoginForm" onSubmit={handleSubmit(async credentials => await login(credentials))}>
      <h2>Log In</h2>
      <FormField name="username" label="Username" {...{register, errors}} validation={{required: 'Please enter your username'}}/>
      <PasswordField name="password" label="Password" {...{register, errors}} validation={{required: 'Please enter your password'}}/>
      <button type="submit">Log in</button> or <Link to="/signup">Sign up</Link>
    </form>
  );
}
