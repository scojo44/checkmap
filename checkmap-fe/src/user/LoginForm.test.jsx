import {render,fireEvent, waitFor} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import LoginForm from './LoginForm'

const mockLogin = vi.fn();

describe('LoginForm Tests', () => {
  beforeEach(() => {
    mockLogin.mockClear();
  });

  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Calls the passed function with the submitted values', async () => {
    const {getByText, getByLabelText} = render(
      <MemoryRouter>
        <LoginForm login={mockLogin}/>
      </MemoryRouter>
    );

    const username = getByLabelText('Username:');
    const password = getByLabelText('Password:');
    const button = getByText('Log in');

    fireEvent.change(username, {target: {value: 'testuser'}});
    fireEvent.change(password, {target: {value: 'secret123'}});
    fireEvent.click(button);

    await waitFor(() => expect(mockLogin).toHaveBeenCalled());
    await waitFor(() => expect(mockLogin).toHaveBeenCalledWith({
      username: 'testuser',
      password: 'secret123'
    }));
  });
});
