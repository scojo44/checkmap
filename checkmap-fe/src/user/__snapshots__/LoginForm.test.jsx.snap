// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginForm Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="LoginForm"
  >
    <h2>
      Log In
    </h2>
    <p
      class="FormField"
    >
      <label
        for="username"
      >
        Username: 
      </label>
      <input
        id="username"
        name="username"
      />
    </p>
    <p
      class="FormField"
    >
      <label
        for="password"
      >
        Password: 
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
    </p>
    <button
      type="submit"
    >
      Log In
    </button>
     or 
    <a
      href="/signup"
    >
      Sign up
    </a>
  </form>
</DocumentFragment>
`;
